### PR TITLE
Address minimal_admin deprecation

### PR DESCRIPTION
**Pull request by jaw6**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

I've been seeing a lot of warnings like this recently:

```
DEPRECATION WARNING: minimal_admin? is deprecated and will be removed from Rails 7.1 (Deprecating ApplicationPolicy#user_admin?, use ApplicationPolicy#user_super_admin?) (called from block (3 levels) in <top (required)> at /home/runner/work/forem/forem/spec/policies/html_variant_policy_spec.rb:10)
```

This is a _little_ misleading... the deprecation warning isn't connected to Rails at all, it's [something we did](https://github.com/forem/forem/pull/16523), but the default messaging from the `deprecate` macro assumes Rails' lingo. 

To summarize:

* We used to have a method `minimal_admin?`, which meant something like "at least `admin` but possibly also `super_admin`"
* We refactored, moved some methods around, such that we preferred to use `Authorizer#any_admin?` 
* We deprecated the old `minimal_admin?` in favor of `user_any_admin?` but aliased it, so we could keep using both
* This PR removes what I believe is the last remaining use of `minimal_admin?`
* Since this seems to be the last use, I am also removing the `alias` and `deprecate`

## Added/updated tests?

- [ ] Yes
- [x] No, and this is why: invisible rename
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by benhalpern — APPROVED** [review]

**Review by dukegreene — APPROVED** [review]
Still looks good when I log in as admin!
